app/src/main/java/org/mozilla/tv/firefox/webrender/WebRenderFragment.kt

```diff
@@ -347,8 +347,6 @@ class WebRenderFragment : EngineViewLifecycleFragment(), Session.Observer {
                 TelemetryIntegration.INSTANCE.browserBackControllerEvent()
             }
             else -> {
-                context!!.webRenderComponents.sessionManager.remove()
-
                 // We can get into this state in two situations:
                 // - (1) The user is on the "home page" and the overlay is visible. In this situation we just want to
```

Comment:
I don't think that we want this (2) case anymore - looking at the history this was written a long time ago, before we had the combined overlay/home tiles. That's what's probably actually breaking this (because it's returns true) - and the true was a result of fixing an auto-kotlin-convert: https://github.com/mozilla-mobile/firefox-tv/blame/8c61df2e2f4fe2678418314accb9e31a79d4a00b/app/src/main/java/org/mozilla/tv/firefox/webrender/WebRenderFragment.kt#L366
If the overlay is not visible and we're on a website, we should just exit, not open the overlay. It actually looks like we should just return false for this whole `else` statement of the when/else, and remove the `if !browserOverlay.isVisible` check. That should fix it because it'll allow the app to handle the back button press.

Reply:
Oh yup, this is a much simpler fix!